app/data/domain/cdoTaskList.js

```diff
@@ -1,14 +1,52 @@
 const CdoTask = require('./cdoTask')
+const { ActionAlreadyPerformedError } = require('../../errors/domain/actionAlreadyPerformed')
 
 class CdoTaskList {
+  /**
+   * @param {Cdo} cdo
+   */
   constructor (cdo) {
     this._cdo = cdo
   }
 
+  static dateStageComplete (stage) {
+    return stage instanceof Date
+  }
+
+  static dateIsInTheFuture (date) {
+    return date.getTime() > Date.now()
+  }
+
   get _stageOneComplete () {
     return this.applicationPackSent.completed
   }
 
+  get _stageTwoComplete () {
```

Comment:
I'm not 100% happy with stageOne, stageTwo etc. but I don;t know how better to represent things. Can we do it directly with task names??